test/Main.test.ts

```diff
@@ -1926,13 +2022,24 @@ describe(`MainP${IMPLEMENTATION} contract`, () => {
           .connect(owner)
           .setPrimeBasket([token0.address, rToken.address], [fp('0.5'), fp('0.5')])
       ).to.be.revertedWith('invalid collateral')
-    })
 
-    it('Should allow to set prime Basket if OWNER', async () => {
-      // Set basket
-      await expect(basketHandler.connect(owner).setPrimeBasket([token0.address], [fp('1')]))
-        .to.emit(basketHandler, 'PrimeBasketSet')
-        .withArgs([token0.address], [fp('1')], [ethers.utils.formatBytes32String('USD')])
+      await expect(
+        indexBH.connect(owner).forceSetPrimeBasket([rsr.address], [fp('1')])
+      ).to.be.revertedWith('invalid collateral')
+      await expect(
+        basketHandler.connect(owner).forceSetPrimeBasket([rsr.address], [fp('1')])
+      ).to.be.revertedWith('invalid collateral')
+
+      await expect(
+        indexBH
+          .connect(owner)
+          .forceSetPrimeBasket([token0.address, rToken.address], [fp('0.5'), fp('0.5')])
+      ).to.be.revertedWith('invalid collateral')
+      await expect(
+        basketHandler
+          .connect(owner)
+          .forceSetPrimeBasket([token0.address, rToken.address], [fp('0.5'), fp('0.5')])
+      ).to.be.revertedWith('invalid collateral')
     })
 
     it('Should revert if target has been changed in asset registry', async () => {
```

Comment:
should this even be allowed?

Reply:
hard to prevent because can break `swapRegistered()` into `unregister()` and `register()`